src/modules/cache-client/auth.ts

```diff
@@ -0,0 +1,50 @@
+import { AxiosInstance, AxiosResponse } from 'axios';
+import { SiweMessage } from 'siwe';
+import { cacheConfigs } from '../../config';
+import { SignerService } from '../signer';
+import { AuthTokens, SiweOptions } from './cache-client.types';
+
+/**
+ * Provides authentication methods to ssi-hub
+ *
+ * @todo Move here authentication parts of `CacheClient`
+ * @todo Make this class top-level module in order to use in application
+ */
+export class SsiAuth {
+  private readonly config: SiweOptions;
+
+  constructor(
+    private signerService: SignerService,
+    private http: AxiosInstance
+  ) {
+    this.config = cacheConfigs()[this.signerService.chainId].auth;
+  }
+
+  /**
+   * Signs in to ssi-hub. Authentication method configurations are looked for in `cacheConfigs`
+   *
+   * @returns authentication tokens
+   */
+  async signIn(): Promise<AxiosResponse<AuthTokens>> {
+    const {
+      data: { nonce },
+    } = await this.http.post<{ nonce: string }>('/login/siwe/initiate');
+    const siweMessage = new SiweMessage({
+      statement: `I accept the Terms of Service: ${this.config.domain}`,
+      nonce,
+      domain: new URL(this.config.domain).host,
+      address: this.signerService.address,
+      uri: `${cacheConfigs()[this.signerService.chainId].url}login/siwe/verify`,
```

Comment:
I think this is potentially a bit brittle as it will have a missing`/` if `url` doesn't end with a `/`. I'm not sure what the better approach is though. Maybe using something like this? 
```suggestion
      const url = cacheConfigs()[this.signerService.chainId].url
      uri: `${url}${url.substring(url.length - 1) === '/' ? '' : '/'} login/siwe/verify`,
```

Reply:
Yes, it is a bit dangerous. I changed it to new URL('/login/siwe/verify', cacheConfigs()[this.signerService.chainId].url).href. This will create correct uri regardless of trailing slash